drivers/i2c/i2c_ite_enhance.c

```diff
@@ -0,0 +1,776 @@
+			 */
+			if (i2c_bus_not_available(dev)) {
+				/* Unlock mutex of i2c controller */
+				k_mutex_unlock(&data->mutex);
+				return -EIO;
+			}
+		}
+
+		msgs->flags |= I2C_MSG_START;
+	}
+
+	for (int i = 0; i < num_msgs; i++) {
+
+		data->widx = 0;
+		data->ridx = 0;
+		data->err = 0;
+		data->msgs = &(msgs[i]);
+		data->addr_16bit = addr;
+
+		if (msgs->flags & I2C_MSG_START) {
+			data->i2ccs = I2C_CH_NORMAL;
+			/* enable i2c interrupt */
+			irq_enable(config->i2c_irq_base);
+		}
+		/* Start transaction */
+		i2c_transaction(dev);
+		/* Wait for the transfer to complete */
+		/* TODO: the timeout should be adjustable */
+		res = k_sem_take(&data->device_sync_sem, K_MSEC(100));
+		/*
+		 * The irq will be enabled at the condition of start or
+		 * repeat start of I2C. If timeout occurs without being
+		 * wake up during suspend(ex: interrupt is not fired),
+		 * the irq should be disabled immediately.
+		 */
+		irq_disable(config->i2c_irq_base);
+		/*
+		 * The transaction is dropped on any error(timeout, NACK, fail,
+		 * bus error, device error).
+		 */
+		if (data->err)
+			break;
+
+		if (res != 0) {
+			data->err = ETIMEDOUT;
+			/* reset i2c port */
+			i2c_reset(dev);
+			printk("I2C ch%d:0x%X reset cause %d\n",
+			       config->port, data->addr_16bit, I2C_RC_TIMEOUT);
+			/* If this message is sent fail, drop the transaction. */
+			break;
+		}
+	}
+
+	/* reset i2c channel status */
+	if (data->err || (msgs->flags & I2C_MSG_STOP)) {
+		data->i2ccs = I2C_CH_NORMAL;
+	}
+	/* Unlock mutex of i2c controller */
+	k_mutex_unlock(&data->mutex);
+
+	return i2c_parsing_return_value(dev);
+}
+
+static void i2c_enhance_isr(void *arg)
+{
+	struct device *dev = (struct device *)arg;
+	struct i2c_enhance_data *data = dev->data;
+	const struct i2c_enhance_config *config = dev->config;
+
+	/* If done doing work, wake up the task waiting for the transfer */
+	if (!i2c_transaction(dev)) {
+		k_sem_give(&data->device_sync_sem);
+		irq_disable(config->i2c_irq_base);
+	}
+}
+
+static int i2c_enhance_init(const struct device *dev)
+{
+	struct i2c_enhance_data *data = dev->data;
+	const struct i2c_enhance_config *config = dev->config;
+	uint8_t *base = config->base;
+	uint32_t bitrate_cfg, offset = 0;
+	int error;
+
+	/* Initialize mutex and semaphore */
+	k_mutex_init(&data->mutex);
+	k_sem_init(&data->device_sync_sem, 0, K_SEM_MAX_LIMIT);
+
+	/* Enable clock to specified peripheral */
+	switch ((uint32_t)base) {
+	case DT_REG_ADDR(DT_NODELABEL(i2c3)):
+		offset = CGC_OFFSET_SMBD;
+		break;
+	case DT_REG_ADDR(DT_NODELABEL(i2c4)):
+		offset = CGC_OFFSET_SMBE;
+		break;
+	case DT_REG_ADDR(DT_NODELABEL(i2c5)):
+		offset = CGC_OFFSET_SMBF;
+		break;
+	}
+
+	volatile uint8_t *reg = (volatile uint8_t *)
+				(IT8XXX2_ECPM_BASE + (offset >> 8));
+	uint8_t reg_mask = offset & 0xff;
+	*reg &= ~reg_mask;
+
+	/* Enable I2C function */
+	/* Software reset */
+	IT8XXX2_I2C_DHTR(base) |= IT8XXX2_I2C_SOFT_RST;
+	IT8XXX2_I2C_DHTR(base) &= 0x7F;
```

Comment:
Use `~IT8XXX2_I2C_SOFT_RST` instead of 0x7F.

Reply:
Done.